src/main/java/software/bernie/techarium/pipe/capability/PipeNetworkManagerCapability.java

```diff
@@ -22,8 +22,10 @@
     @Override
     public void tick(ServerWorld world) {
         this.world = world;
+        if (world.getGameTime()%5 > 0)
```

Comment:
What's the point of this exactly?

Reply:
Doing pipe calculations only 4 times a second => Reduce lagg and powers of 2 are awesome.
Also ycar said that the pipes are too fast and Ycar and I decided to do transportation only 4 times a second, because it is simpler or something. Looks like you missed some messages

Reply:
Hm. We need to figure out a proper speed system.

Reply:
Upgrades should change the amount transported anf 8 though about something: if we check gametime%5 == uuid.someIntFunction()%5 we could smooth out the calculations further, so that there aren't any lagg spikes every 5 ticks